app/src/main/java/org/schabi/newpipe/util/ReleaseUtil.kt

```diff
@@ -0,0 +1,160 @@
+            } catch (e: CertificateException) {
+                showRequestError(app, e, "Certificate error")
+                return ""
+            }
+
+            return try {
+                val md = MessageDigest.getInstance("SHA1")
+                val publicKey = md.digest(x509cert.encoded)
+                byte2HexFormatted(publicKey)
+            } catch (e: NoSuchAlgorithmException) {
+                showRequestError(app, e, "Could not retrieve SHA1 key")
+                ""
+            } catch (e: CertificateEncodingException) {
+                showRequestError(app, e, "Could not retrieve SHA1 key")
+                ""
+            }
+        }
+
+    private fun byte2HexFormatted(arr: ByteArray): String {
+        val str = StringBuilder(arr.size * 2)
+        for (i in arr.indices) {
+            var h = Integer.toHexString(arr[i].toInt())
+            val l = h.length
+            if (l == 1) {
+                h = "0$h"
+            }
+            if (l > 2) {
+                h = h.substring(l - 2, l)
+            }
+            str.append(h.uppercase())
+            if (i < arr.size - 1) {
+                str.append(':')
+            }
+        }
+        return str.toString()
+    }
+
+    /**
+     * Method to compare the current and latest available app version.
+     * If a newer version is available, we show the update notification.
+     *
+     * @param versionName    Name of new version
+     * @param apkLocationUrl Url with the new apk
+     * @param versionCode    Code of new version
+     */
+    fun compareAppVersionAndShowNotification(
+        versionName: String,
+        apkLocationUrl: String?,
+        versionCode: Int
+    ) {
+        if (BuildConfig.VERSION_CODE >= versionCode) {
+            return
+        }
+        val app = App.getApp()
+
+        // A pending intent to open the apk location url in the browser.
+        val intent = Intent(Intent.ACTION_VIEW, apkLocationUrl?.toUri())
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        val pendingIntent = PendingIntent.getActivity(app, 0, intent, 0)
+        val channelId = app.getString(R.string.app_update_notification_channel_id)
+        val notificationBuilder = NotificationCompat.Builder(app, channelId)
+            .setSmallIcon(R.drawable.ic_newpipe_update)
+            .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+            .setContentIntent(pendingIntent)
+            .setAutoCancel(true)
+            .setContentTitle(app.getString(R.string.app_update_notification_content_title))
+            .setContentText(
+                app.getString(R.string.app_update_notification_content_text)
+                        + " " + versionName
+            )
+        val notificationManager = NotificationManagerCompat.from(app)
+        notificationManager.notify(2000, notificationBuilder.build())
+    }
+
+    private fun showRequestError(app: App, e: Exception, request: String) {
```

Comment:
Same as my comment above

Reply:
This is only for the `certificateSHA1Fingerprint` getter, to shorten it and make it a bit more readable. It's supposed to be here :)

Reply:
Right, keep it here ;-)